pkg/ingester/client/cortex.proto

```diff
@@ -34,6 +34,7 @@ message WriteRequest {
     RULE = 1;
   }
   SourceEnum Source = 2;
+  repeated MetricMetadata metadata = 3 [(gogoproto.nullable) = true];
```

Comment:
the use of `gogoproto.nullable` is to allow for a slice of pointers, which will help reduce allocations (since we're no longer using a custom type)